pysat/utils/_core.py

```diff
@@ -420,45 +357,128 @@ def load_netcdf4(fnames=None, strict_meta=False, file_format=None,
+                                else:
+                                    time_var = pds.to_datetime(1E6*time_var)
+                            new_index = time_var
+                            new_index_name = index_name
+                        else:
+                            # using integer indexing
+                            new_index = np.arange(loop_lim*step_size,
+                                                  dtype=int) % step_size
+                            new_index_name = 'index'
+                        # load all data into frame
+                        loop_frame = pds.DataFrame(loop_dict[obj_key_name])
+                        # del loop_frame['dimension_1']
+                        # break massive frame into bunch of smaller frames
+                        for i in np.arange(loop_lim, dtype=int):
+                            loop_list.append(loop_frame.iloc[step_size*i:step_size*(i+1), :])
+                            loop_list[-1].index = new_index[step_size*i:step_size*(i+1)]
+                            loop_list[-1].index.name = new_index_name
+
+                        # add 2D object data, all based on a unique dimension
+                        # within netCDF, to loaded data dictionary
+                        loadedVars[obj_key_name] = loop_list
+                        del loop_list
+
+                # prepare dataframe index for this netcdf file
+                time_var = loadedVars.pop(epoch_name)
+
+                # convert from GPS seconds to seconds used in pandas (unix time,
+                # no leap)
+                # time_var = convert_gps_to_unix_seconds(time_var)
+                if file_format == 'NETCDF4':
+                    loadedVars[epoch_name] = \
+                        pds.to_datetime((1E6 * time_var).astype(int))
+                else:
+                    loadedVars[epoch_name] = \
+                        pds.to_datetime((time_var * 1E6).astype(int))
+                running_store.append(loadedVars)
+                running_idx += len(loadedVars[epoch_name])
+
+                if strict_meta:
+                    if saved_mdata is None:
+                        saved_mdata = copy.deepcopy(mdata)
+                    elif (mdata != saved_mdata):
+                        raise ValueError(' '.join(('Metadata across filenames',
+                                                   'is not the same.')))
+
+        # combine all of the data loaded across files together
+        out = []
+        for item in running_store:
+            out.append(pds.DataFrame.from_records(item, index=epoch_name))
+        out = pds.concat(out, axis=0)
+    else:
+        out = xr.open_mfdataset(fnames, combine='by_coords')
+        for key in out.variables.keys():
+            meta_dict = {}
+            for nc_key in out.variables[key].attrs.keys():
+                meta_dict[nc_key] = out.variables[key].attrs[nc_key]
+            mdata[key] = meta_dict
```

Comment:
This is the xarray equivalent of the above load routines.  The multifile dataset should handle stitching files together, and then it's just a matter of exporting the metadata into a pysat metadata object.  I'm not sure if I'm capturing everything pysat needs for metadata.  Comments welcome.